src/main/java/hudson/plugins/ansicolor/AnsiAttributeElement.java

```diff
@@ -26,14 +25,20 @@
     String name;
     String attributes;
 
-    public static interface Emitter {
-        public void emitHtml(String html);
+    public interface Emitter {
+        void emitHtml(@Nonnull String html);
+
         /**
-         * Called when SGR0 (Set Graphics Rendition 0) reset is encountered in the stream, but no tags have been opened
-         * since its last occurrence. If you are using the output of AnsiOutputStream directly, you probably don't need
-         * to implement this method since the escape sequence will be transparently filtered from the output.
+         * Emit invisible ANSI sequence
+         * <p>
+         * Used for emitting output for sequences like:
+         * <li>SGR0 (Set Graphics Rendition 0)
+         * <li>CSI (Control Sequence Introducer)
+         * <p>
+         * In particular called when SGR0 reset is encountered in the stream, but no tags have been opened since its last occurrence. If you are using the output of AnsiOutputStream directly, you
+         * probably don't need to implement this method since the escape sequence will be transparently filtered from the output.
          */
-        default void emitRedundantReset() { }
+        default void emitInvisibleSequence() { }
```

Comment:
Note: This breaks binary compatibility. We could avoid it by keeping `emitRedundantReset` and having it call `emitInvisibleSequence` internally, but [it doesn't look like `emitRedundantReset` was ever used by any other plugins](https://github.com/search?utf8=%E2%9C%93&q=user%3Ajenkinsci+emitRedundantReset&type=Code), so I think the change is fine as-is.

Reply:
> 
> 
> Note: This breaks binary compatibility.

Are there other plugins extending `ansicolor-plugin`?

Reply:
[updatebot plugin](https://github.com/jenkinsci/updatebot-plugin/blob/4c8db79c112b670320b23596b01382f17a2de94c/src/main/java/hudson/plugins/ansicolor/AnsiHelper.java#L34) implements `AnsiAttributeElement.Emitter` (the usage looks weird, it probably _shouldn't_ be extending ansicolor), but the plugin looks obsolete and [has less than 50 installs](https://plugins.jenkins.io/updatebot/), so I think it can be ignored.